src/section_list_cache.rs

```diff
@@ -72,16 +71,20 @@ impl SectionListCache {
             .entry(prefix)
             .or_insert_with(HashMap::new)
             .entry(list)
-            .or_insert_with(HashMap::new)
+            .or_insert_with(BTreeMap::new)
             .insert(pub_id, sig);
         self.update_lists_cache(our_section_size);
     }
 
     /// Returns the currently signed group list for `prefix` along with a quorum of signatures.
-    // TODO: Remove this when the method is used in production
-    #[cfg(feature="use-mock-crust")]
-    pub fn get_signatures(&self, prefix: Prefix<XorName>) -> Option<&(SectionList, Signatures)> {
-        self.lists_cache.get(&prefix)
+    pub fn get_signed_list(&self, prefix: &Prefix<XorName>) -> Option<SignedSectionList> {
+        // TODO: why not store this type directly? And do we need to clone?
```

Comment:
It will probably make sense to use it directly in `lists_cache`. I decided to use another map in `signatures`, as it makes it easier to add new signatures without first searching if we already have a list in the cache.